test/foundry/integration/IntegrationBase.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import {Test} from "forge-std/Test.sol";
+import {Allo, IAllo} from "contracts/core/Allo.sol";
+import {ITransparentUpgradeableProxy} from "./ITransparentUpgradeableProxy.sol";
+import {IOwnable} from "./IOwnable.sol";
+
+abstract contract IntegrationBase is Test {
+    address public constant ALLO_PROXY = 0x1133eA7Af70876e64665ecD07C0A0476d09465a1;
+    address public constant BICONOMY_FORWARDER = 0x84a0856b038eaAd1cC7E297cF34A7e72685A8693;
+
+    address public alloAdmin;
+    address public owner;
+    address public registry;
+    address public treasury;
+
+    uint256 public percentFee;
+    uint256 public baseFee;
+
+    function setUp() public virtual {
+        vm.createSelectFork(vm.rpcUrl("mainnet"), 20289932);
+
+        // Get the current protocol variables
+        // TODO: call admin() instead of reading storage 
+        // alloAdmin = ITransparentUpgradeableProxy(ALLO_PROXY).admin();
```

Comment:
this doesnt work?

Reply:
Nope, for some reason it fails. I will try debug it again.

Reply:
AFAIK `admin()` is only callable by the admin itself. So, if you are the admin you can call `admin()` to see that you are the admin. If you are not the admin you won't be able to know who the admin is, but you can check that you are not the admin. Super intuitive.

The admin is usually a ProxyAdmin contract which address should be known and which can be an admin of several proxies. In this case the ProxyAdmin is [0x758b87af7fdB4783f848a1dDEa1F025dC48B9858](https://etherscan.io/address/0x758b87af7fdB4783f848a1dDEa1F025dC48B9858) and has `0x34d82D1ED8B4fB6e6A569d6D086A39f9f734107E` as owner.

Reply:
If the ProxyAdmin address is not known, it can usually be found by looking at the contract creation transaction logs. I don't know if there is a better way.

Reply:
I know on hardhat we could regenerate the our zeppelin file , is there no way to do that for foundry ?

Reply:
Sorry!, missed this comment. 
I'm not sure if that is possible. Anyway, we can query the admin reading storage so is not a big deal.